src/master-playlist-controller.js

```diff
@@ -820,23 +826,21 @@ export class MasterPlaylistController extends videojs.EventTarget {
       return this.mediaSource.endOfStream('network');
     }
 
-    // Blacklist this playlist
-    currentPlaylist.excludeUntil = Date.now() + BLACKLIST_DURATION;
+    if (!this.masterPlaylistLoader_.isFinalRendition_()) {
```

Comment:
Really minor, and is mostly style preference, but if you change to use a check for it being the final rendition, and handling that case and returning, only one line needs to be indented, and the standard case (of blacklisting) can follow.